### ISSUE TITLE
PyTorch OP List(P0)

### ISSUE THREAD

**Issue by PaParaZz1** [opendilab/DI-treetensor#1]
reference: https://pytorch.org/docs/1.8.0/torch.html
#### common
* [x] numel
* [x] cpu
* [x] cuda
* [x] to
#### Creation Ops
* [x] torch.zeros_like
* [x] torch.randn_like
* [x] torch.randint_like
* [x] torch.ones_like
* [x] torch.full_like
* [x] torch.empty_like
* [x] torch.zeros
* [x] torch.randn
* [x] torch.randint
* [x] torch.ones
* [x] torch.full
* [x] torch.empty
#### Indexing, Slicing, Joining, Mutating Ops
* [x] cat
* [x] chunk
* [ ] gather
* [x] index_select
* [x] masked_select
* [x] reshape
* [ ] scatter
* [x] split
* [x] squeeze
* [x] stack
* [ ] tile
* [ ] unbind
* [x] unsqueeze
* [x] where
##### Pointwise Ops
* [x] add
* [x] sub
* [x] mul
* [x] div
* [x] pow
* [x] neg
* [x] abs
* [x] sign
* [x] floor
* [x] ceil
* [x] round
* [x] sigmoid
* [x] clamp
* [x] exp
* [x] exp2
* [x] sqrt
* [x] log
* [x] log10
* [x] log2

##### Reduction Ops
* [ ] argmax
* [ ] argmin
* [x] all
* [x] any
* [x] max
* [x] min
* [x] dist
* [ ] logsumexp
* [x] mean
* [ ] median
* [x] norm
* [ ] prod
* [x] std
* [x] sum
* [ ] unique
##### Comparison Ops
* [ ] argsort
* [x] eq
* [x] ge
* [x] gt
* [x] isfinite
* [x] isinf
* [x] isnan
* [x] le
* [x] lt
* [x] ne
* [ ] sort
* [ ] topk
##### Other Ops
* [ ] cdist
* [x] clone
* [ ] flip
#### BLAS and LAPACK Ops
* [ ] addbmm
* [ ] addmm
* [ ] bmm
* [x] dot
* [x] matmul
* [x] mm

**Comment by HansBug** [comment]
Add the todo checkboxes into the issue.

**Comment by PaParaZz1** [comment]
#### In-place OP List:
##### In-place random sampling
- normal_
- log_normal_
- random_
- uniform_

##### Calculation
- abs_
- add_
- addbmm_
- addmm_
- ceil_
- clamp_
- cos_
- div_
- exp_
- floor_
- log_
- log2_
- mul_
- neg_
- pow_
- round_
- sigmoid_
- sign_
- sin_
- sqrt_
- square_
- squeeze_
- sub_
- tan_
- tanh_
- unsqueeze_
##### Other
- requires_grad_

**Comment by HansBug** [comment]
All the ops has been supported by dynamic features.
